### PR TITLE
bug 1159295 - Move auto-closing change sets to middleware

### PR DESCRIPTION
**Pull request by jwhitlock**
The root cause of [bug 1159295](https://bugzilla.mozilla.org/show_bug.cgi?id=1159295) (500 error when deleting a specification) is that the auto-created changesets only allowed a single instance to be modified, and deleting the specification triggered a cascading deletion of related sections.  By moving the logic for closing auto-created changesets to the middleware's response handler, multiple instance changes can occur in a single auto-created changeset.  This makes cascading deletes work, as well as simplifies the logic for POSTs to the view_feature endpoint.

### CODE REVIEW

**Comment by renoirb** [comment]
Thanks for the explaination.

**Comment by jwhitlock** [comment]
Rebased on master

File: webplatformcompat/tests/test_viewsets.py [hunk 1]
```diff
@@ -272,10 +272,74 @@ def test_delete_in_changeset(self, mock_update):
...
+        mock_update.side_effect = Exception("not called")
         response = self.client.delete(url)
         self.assertEqual(204, response.status_code, response.content)
         self.assertFalse(Browser.objects.filter(pk=browser.pk).exists())
-        mock_update.assert_not_called()
+
+
+class TestCascadeDelete(APITestCase):
+    def setUp(self):
+        self.login_user(groups=["change-resource", "delete-resource"])
```
Comment (jezdez) [inline webplatformcompat/tests/test_viewsets.py]: super call missing

**Comment by jezdez** [comment]
r+

**Pull request merged**